### DISCUSSION TITLE
VipsJpeg: Maximum supported image dimension is 65500 pixels getting this error after using arrayjoin. [proposed Label] question

### DISCUSSION THREAD

**Discussion by Rochan2001** [libvips/libvips#2249] [Q&A] [answered]
I have used jcupitt's code from stackoverflow to join jpeg images and convert them into a single tiff file.
This is the code:

```python
#!/usr/bin/env python3

import pyvips

tiles_across = 142
tiles_down = 150

tiles = [pyvips.Image.new_from_file(f"{x}_{y}.jpeg", access="sequential")
         for y in range(tiles_down) for x in range(tiles_across)]
im = pyvips.Image.arrayjoin(tiles, across=tiles_across)

im.write_to_file("x.jpg")
```

I have 105,710 jpeg images in my last folder after using dzsave. Is there a workaround for this problem because I think it's jpegs limitation which is causing this problem.

**Comment by jcupitt** [comment]
Hi, you'll need to use a different file format -- JPEG images cannot be larger than 65535 pixels on either axis.

Tiled TIFF with JPEG compression would be a good choice.

**Reply by Rochan2001** [reply]
thank you for replying, it's working with png images.

**Reply by jcupitt** [reply]
I'm glad it's working. PNG is slow and the files will be very large.

**Reply by Rochan2001** [reply]
how can i use Tiled tiff with JPEG compression with dzsave then?

**Reply by jcupitt** [reply]
You're not using `dzsave` in your sample program above. You're writing a very large, single image.

**Reply by Rochan2001** [reply]
sorry i'm a noob at this, can you explain in detail about using tiled tiff with jpeg compression to solve my problem?

**Comment by jcupitt** [comment, marked as answer]
To save a huge tiled jpeg tiff image, change your last line to be:

```
im.write_to_file("x.tif", tile=True, compression="jpeg")
```

**Reply by Rochan2001** [reply]
is this same as using tiffsave?

**Reply by jcupitt** [reply]
Yes, if you call `write_to_file` with a filename that ends with `".tif"`, it'll call `tiffsave` for you.

**Reply by Rochan2001** [reply]
thank you,i got it.
